### ISSUE TITLE
Approve Form Causes Major Version to update ID, and Wizard On Stage Doesn't reflect new ID or content

### ISSUE THREAD

**Issue by benallenallen** [tsgrp/HPI#1988]
On HBase, when a user approves a form and/or a controlled document, the document is made a 1.0 MAJOR version. The process that is currently used to do this copies over the previous version's row (ex: xxxxxx-0.1) to a new row (ex: xxxxxx-1.0). The copy process blindly takes the value of the content_application/pdf (and/or all other content columns) and copies it over to the new row. The next step in most cases then deletes the minor versions of the form/controlled document, which WILL delete the underlying content from the repository. Then when a user asks for the content of the 1.0, it looks for the content for the 0.1, which is deleted. The "fix" could be done in various ways:

1. don't delete the content of the minor version document in the delete
1. manually copy the content over to the new row with a new value so that the delete doesn't actually delete the content that we want.

Also need to address the issue that it seems like the stage doesn't refresh with the new ID of the form. It looks like the IDs "might" be getting updated in the URL bar, but it also needs to trigger a refresh of the stage with the new document and container (same ID for both)

HBaseVersionManager.java
![image](https://user-images.githubusercontent.com/4850976/68409848-6cf00400-014d-11ea-81b0-a83272bef385.png)

![issueGif3](https://user-images.githubusercontent.com/4850976/68410557-847bbc80-014e-11ea-8f28-96ef8ae39239.gif)

**Comment by benallenallen** [comment]
A "workaround" exists in configuring it to NOT delete the minor versions does "fix" this.

**Comment by benallenallen** [comment]
Correction to the above comment - NOT deleting the minor versions does not fix this since the last version (ex: 0.3) is ALWAYS "deleted" since it just rebrands the 0.3 to be 1.0.

**Comment by benallenallen** [comment]
something along these lines in the HBaseVersionManager#promoteToMajorVersion method:

```
			// Perform a complete copy of the row in hbase
			NavigableMap<byte[], NavigableMap<byte[], byte[]>> familyQualifierMap = result.getNoVersionMap();
	        for (byte[] familyBytes : familyQualifierMap.keySet()) {
	            NavigableMap<byte[], byte[]> qualifierMap = familyQualifierMap.get(familyBytes);
	
	            for (byte[] qualifier : qualifierMap.keySet()) {
	                put.addColumn(familyBytes, qualifier, qualifierMap.get(qualifier));
	                
	                //TODO: if this is a 'content_xxxx/xxxx' (where xxxx/xxxx is a mimetype) field, we need to grab the content to be able to "copy" it to the new version. Make sure we copy all content (renditions/primary content)
	                /*
	                  	//get the existing content for this document/mimetype combination
	                  	EnhancedObjectContent eoc = HBaseUtil.getContentFromStorage(objectId, xxxxx/xxxxx, result)
	                  	
	                  	//store the content on the underlying storage (might be external NAS, S3, HBase bytes directly, etc)
						String contentLocation = store.put(eoc, objectId, encryptionKeyBytes);
						
						//if the content storage impl ultimately returned us a string of the path, we need to set the content attribute with the path:
						//ex: content_application/pdf would be stored as /2019/07/02/11/15/GUID-0.1.pdf
						if (StringUtils.isNotBlank(contentLocation)) {
							// Save the content_MIMETYPE property
							byte[] contentPropertyBytes = HBaseUtil.convertToByteArray(new String[]{contentLocation}, HBaseConstants.PROPERTY_TYPE_STRING);
							put.addColumn(Bytes.toBytes(HBaseConstants.COL_FAM_CONTENT), Bytes.toBytes(HBaseConstants.PREFIX_CONTENT + mimeType), contentPropertyBytes);
						}
	                 */
	            }            
	        }  
```

**Comment by benallenallen** [comment]
This was "mostly" fixed in OC Commit #23849, but the stage seems to have an issue where the container isn't being refreshed with the updated actions/properties.

**Comment by benallenallen** [comment]
![issueGif3](https://user-images.githubusercontent.com/4850976/68871942-d7132680-06c2-11ea-85aa-a31c55b97c13.gif)

Seems like a stage issue now... the /action/evaluate call that is triggered for the stage does not reflect the new ID... it seems like it is depending on the |CURRENT thing to have resolved the new ID, but that only happens after the "complete approval" modal is closed. You can see that the 'Stage Info' does properly refresh when the modal is closed with the proper info, but the call to evaluate is still using the old ID and occurs BEFORE the modal is closed and updated with the new ID somehow? Race Condition?

**Comment by benallenallen** [comment]
The final issue was addressed in OCMS commit #24312 by skovacinski
